corehq/apps/ota/decorators.py

```diff
@@ -0,0 +1,39 @@
+from corehq import toggles
+from corehq.apps.users.decorators import require_permission
+from corehq.apps.users.models import Permissions
+
+from dimagi.utils.couch.cache.cache_core import get_redis_client
+
+from functools import wraps
+
+from django.http import HttpResponseForbidden
+
+ORIGIN_TOKEN_HEADER = 'HTTP_X_COMMCAREHQ_ORIGIN_TOKEN'
```

Comment:
Is there magic somewhere that's turning `X-CommCareHQ-Origin-Token` into this? I see the pattern elsewhere but don't understand how it works.

Reply:
Yeah, I was also really confused (am still somewhat confused)

I think Django didn't support a clean headers bucket until a very recent version (that we aren't on, I tried to use it), and in the mean time their approach was to pile a bunch of other request metadata into a single dict, and prepend all of the HTTP headers into that bucket with "HTTP_", upcase them, and replace all non[a-zA-Z] chars with `_`. 

Someone who's more familiar with django probably has a fun story about why this definitely made sense at some point and has been a massive painpoint for everyone else for the last 10 years.